libs/api/domains/license-service/src/lib/licenseService.service.ts

```diff
@@ -1,78 +1,84 @@
+import { UserLicensesResponse } from './dto/UserLicensesResponse.dto'
+
 import {
-  GenericLicenseTypeType,
-  GenericLicenseType,
+  GenericLicenseFetchResult,
+  GenericLicenseMapper,
+  GenericLicenseOrganizationSlug,
+  LicenseTypeKey,
+  GenericLicenseUserdata,
   GenericUserLicenseFetchStatus,
+  GenericUserLicensePkPassStatus,
   GenericUserLicenseStatus,
+  LicenseTokenData,
   PkPassVerification,
-  GenericUserLicensePkPassStatus,
-  GenericLicenseOrganizationSlug,
-  GenericLicenseUserdata,
-  GenericLicenseFetchResult,
-  LICENSE_MAPPER_FACTORY,
-  GenericLicenseMapper,
-  DEFAULT_LICENSE_ID,
 } from './licenceService.type'
-import { Locale } from '@island.is/shared/types'
 import {
-  LicenseClient,
-  LicenseClientService,
-  LicenseType,
-} from '@island.is/clients/license-client'
-import { AVAILABLE_LICENSES } from './licenseService.module'
-import {
-  UserLicensesResponse,
-  GenericUserLicense,
-} from './graphql/genericLicense.model'
+  AVAILABLE_LICENSES,
+  DEFAULT_LICENSE_ID,
+  LICENSE_MAPPER_FACTORY,
+  LICENSE_SERVICE_CACHE_MANAGER_PROVIDER,
+  TOKEN_SERVICE_PROVIDER,
+} from './licenseService.constants'
+import { TokenService } from './services/token.service'
 
 const LOG_CATEGORY = 'license-service'
 
 export type GetGenericLicenseOptions = {
-  includedTypes?: Array<GenericLicenseTypeType>
-  excludedTypes?: Array<GenericLicenseTypeType>
+  includedTypes?: Array<LicenseTypeKey>
+  excludedTypes?: Array<LicenseTypeKey>
   force?: boolean
   onlyList?: boolean
 }
+
+const { randomUUID } = new ShortUniqueId({ length: 10 })
+
 @Injectable()
 export class LicenseServiceService {
   constructor(
     @Inject(LOGGER_PROVIDER) private logger: Logger,
+    @Inject(LICENSE_SERVICE_CACHE_MANAGER_PROVIDER)
+    private readonly cacheManager: CacheManager,
+    @Inject(TOKEN_SERVICE_PROVIDER)
+    private readonly tokenService: TokenService<LicenseTokenData>,
     private readonly licenseClient: LicenseClientService,
     private readonly cmsContentfulService: CmsContentfulService,
+
     @Inject(LICENSE_MAPPER_FACTORY)
     private readonly licenseMapperFactory: (
-      type: GenericLicenseType,
+      type: LicenseType,
     ) => Promise<GenericLicenseMapper | null>,
   ) {}
 
   private async getOrganization(
     slug: GenericLicenseOrganizationSlug,
     locale: Locale,
   ) {
-    const organization = await this.cmsContentfulService.getOrganization(
-      slug,
-      locale,
-    )
-
-    return organization
+    return this.cmsContentfulService.getOrganization(slug, locale)
   }
-
-  //backwards compatibilty hax
-  private mapLicenseType = (type: GenericLicenseType) =>
-    this.licenseClient.getClientByLicenseType(
-      type === GenericLicenseType.DriversLicense
-        ? LicenseType.DrivingLicense
-        : (type as unknown as LicenseType),
-    )
+  private mapLicenseType = <Type extends LicenseType>(type: LicenseType) =>
+    this.licenseClient.getClientByLicenseType<Type>(type)
```

Comment:
~Can get rid of this function now. `this.licenseClient.getClientByLicenseType()` is clearer then `this.mapLicenseType`.